Inc/HALALMock/Models/PinModel/Pin.hpp

```diff
@@ -78,6 +78,65 @@ enum PinState{
 	ON
 };
 
+enum PinType {
```

Comment:
```suggestion
enum class PinType {
```

Reply:
regular enums can be mix and matched, for example, I could have another enum somewhere else like:
```c++
enum FDCANMode {
    Loopback,
    External,
};
```
And then somewhere do:
```c++
PinType pin_type = PinMode::DigitalOutput;
if (pin_type == FDCANMode::Loopback) {
    \\ ...
}
```
And this would compile, which is not optimal, because the enum is implicitly casted to a number and compared. `enum class` avoids this issue.